server/app/routes/api/users.js

```diff
@@ -17,73 +25,45 @@ router.get('/', function (req, res) {
 })
 
 // only available to admin and user themself?
-router.get('/:id', function (req, res) {
-  var id = req.params.id;
-  mongoose.model('User')
-  .findById(id)
-  .then(function(user) {
-    // is it a promise?? //TW nope
-    return user.sanitize()
-  })
-  .then(function(sanitizedUser) {
-    res.json(sanitizerUser)
-  })
-  .then(null, next);
+router.get('/:userId', function (req, res) {
+  res.json(req.reqUser.sanitize())
 })
 
-// TW if you wanted to promisify user.sanitize 
-// Promise.resolve(user.sanitize())
-// .then(function(user){
-
-// })
-
 // admin and user
 // is this already taken care of in orders route??
-router.get('/:id/orders', function (req, res) {
-  var id = req.params.id;
-  mongoose.model('User')
-  .findById(id)
-  .then(function(user) {
-    return user.findOrders()
-  })
+router.get('/:userId/orders', function (req, res, next) {
+  req.reqUser.findOrders()
   .then(function(orders) {
     res.json(orders)
   })
-  .then(null, next);
+  .then(null, next)
 })
 
 // post a new user
-router.post('/', function (req, res) {
-  mongoose.model('User')
-  .create(req.body)
+router.post('/', function (req, res, next) {
+  User.create(req.body)
   .then(function(newUser){
-    res.json(newUser)
+    res.status(201).json(newUser)
   })
   .then(null, next);
 })
 
 // update a user
-router.put('/:id', function (req, res) {
-  var id = req.params.id
-  //TW refactor with set
-  mongoose.model('User') 
-  .findByIdAndUpdate(id, {$set: req.body}, {new: true, runValidators: true})
+router.put('/:userId', function (req, res, next) {
+  req.reqUser.set(req.body)
```

Comment:
maybe an argument for using update rather than set & save in this case? (not happy about having secrets outside the DB, but it's probably not the end of the world)